docs/collections/electric-collection.md

```diff
@@ -0,0 +1,217 @@
+---
+title: Electric Collection
+---
+
+# Electric Collection
+
+Electric collections provide seamless integration between TanStack DB and ElectricSQL, enabling real-time data synchronization with your Postgres database through Electric's sync engine.
+
+## Overview
+
+The `@tanstack/electric-db-collection` package allows you to create collections that:
+- Automatically sync data from Postgres via Electric shapes
+- Support optimistic updates with transaction confirmation and automatic rollback on errors
+- Handle persistence through customizable mutation handlers
+
+## Installation
+
+```bash
+npm install @tanstack/electric-db-collection @tanstack/db
+```
+
+## Basic Usage
+
+```typescript
+import { createCollection } from '@tanstack/db'	
+import { electricCollectionOptions } from '@tanstack/electric-db-collection'
+
+const todosCollection = createCollection(
+  electricCollectionOptions({
+    shapeOptions: {
+      url: 'https://example.com/v1/shape',
+      params: {
+        table: 'todos',
+      },
+    },
+    getKey: (item) => item.id,
+  })
+)
+```
+
+## Configuration Options
+
+The `electricCollectionOptions` function accepts the following options:
+
+### Required Options
+
+- `shapeOptions`: Configuration for the ElectricSQL ShapeStream
+- `getKey`: Function to extract the unique key from an item
+
+### Shape Options
+
+- `url`: The URL to your Electric sync service
+- `params`: Shape parameters including:
+  - `table`: The database table to sync
+  - `where`: Optional WHERE clause for filtering (e.g., `"status = 'active'"`)
+  - `columns`: Optional array of columns to sync
+
+### Collection Options
+
+- `id`: Unique identifier for the collection
+- `schema`: Schema for validating items (any Standard Schema compatible schema)
+- `sync`: Custom sync configuration
+
+### Persistence Handlers
+
+- `onInsert`: Handler called before insert operations
+- `onUpdate`: Handler called before update operations  
+- `onDelete`: Handler called before delete operations
+
+## Persistence Handlers
+
+You can define handlers that are called when mutations occur, for instance to persist changes to your backend. Such handlers **must return a transaction ID** (`txid`) to track when the mutation has been synchronized back from Electric:
```

Comment:
Txid is recommended but there are already other ways of doing this and we'll be adding more

https://github.com/TanStack/db/issues/402

Reply:
It's good to explain why this is ideal for sync matching